pkg/controllers/provisioning/batch.go

```diff
@@ -0,0 +1,109 @@
+/*
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package provisioning
+
+import (
+	"context"
+	"time"
+
+	"github.com/awslabs/karpenter/pkg/apis/provisioning/v1alpha5"
+	v1 "k8s.io/api/core/v1"
+)
+
+var (
+	MaxBatchDuration = time.Second * 10
+	MinBatchDuration = time.Second * 1
+	// MaxPodsPerBatch limits the number of pods we process at one time to avoid using too much memory
+	MaxPodsPerBatch = 2_000
+)
+
+type Batch struct {
+	Key         string
+	Constraints *v1alpha5.Constraints
```

Comment:
Is there any way for the `Batch` struct not to "know" about `Constraints`? AFAICT it's just acting as a sort of pass-through (while I'm not suggesting this change, as a thought experiment, it seems like `Batch` could just have an inscrutable `interface{}` here just as well, since it doesn't care about the actual contents? Another way to look at it is that having `Constraints` "in" the `Batch` struct makes it seems like the methods in this file are going to do more than just passively add pods to a channel, but instead be more involved in determining which pods should be added to which channels?

Reply:
I like this idea. Check out the new version that's type agnostic and supports range.